plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -152,111 +227,215 @@ def get_user_access_level(self, members, gitlab_user_id):
+            ['state', 'present', ['access_level', 'gitlab_users_access'], True],
         ],
         supports_check_mode=True,
     )
 
     if not HAS_PY_GITLAB:
         module.fail_json(msg=missing_required_lib('python-gitlab', url='https://python-gitlab.readthedocs.io/en/stable/'), exception=GITLAB_IMP_ERR)
 
+    access_level_int = {
+        'guest': gitlab.GUEST_ACCESS,
+        'reporter': gitlab.REPORTER_ACCESS,
+        'developer': gitlab.DEVELOPER_ACCESS,
+        'maintainer': gitlab.MAINTAINER_ACCESS,
+        'owner': gitlab.OWNER_ACCESS
+    }
+
     gitlab_group = module.params['gitlab_group']
-    gitlab_user = module.params['gitlab_user']
     state = module.params['state']
     access_level = module.params['access_level']
+    purge_users = module.params['purge_users']
 
-    # convert access level string input to int
-    if access_level:
-        access_level_int = {
-            'guest': gitlab.GUEST_ACCESS,
-            'reporter': gitlab.REPORTER_ACCESS,
-            'developer': gitlab.DEVELOPER_ACCESS,
-            'maintainer': gitlab.MAINTAINER_ACCESS,
-            'owner': gitlab.OWNER_ACCESS
-        }
-
-        access_level = access_level_int[access_level]
+    if purge_users:
+        purge_users = [access_level_int[level] for level in purge_users]
 
     # connect to gitlab server
     gl = gitlabAuthentication(module)
 
     group = GitLabGroup(module, gl)
 
-    gitlab_user_id = group.get_user_id(gitlab_user)
     gitlab_group_id = group.get_group_id(gitlab_group)
 
     # group doesn't exist
     if not gitlab_group_id:
         module.fail_json(msg="group '%s' not found." % gitlab_group)
 
-    # user doesn't exist
-    if not gitlab_user_id:
-        if state == 'absent':
-            module.exit_json(changed=False, result="user '%s' not found, and thus also not part of the group" % gitlab_user)
-        else:
-            module.fail_json(msg="user '%s' not found." % gitlab_user)
-
-    members = group.get_members_in_a_group(gitlab_group_id)
-    is_user_a_member = group.is_user_a_member(members, gitlab_user_id)
-
-    # check if the user is a member in the group
-    if not is_user_a_member:
-        if state == 'present':
-            # add user to the group
-            if not module.check_mode:
-                group.add_member_to_group(gitlab_user_id, gitlab_group_id, access_level)
-            module.exit_json(changed=True, result="Successfully added user '%s' to the group." % gitlab_user)
-        # state as absent
-        else:
-            module.exit_json(changed=False, result="User, '%s', is not a member in the group. No change to report" % gitlab_user)
-    # in case that a user is a member
+    members = []
+    if module.params['gitlab_user'] is not None:
+        gitlab_users_access = []
+        gitlab_users = module.params['gitlab_user']
+        for gl_user in gitlab_users:
+            gitlab_users_access.append({'name': gl_user, 'access_level': access_level_int[access_level]})
```

Comment:
Here on the other hand, it can be that `not access_level` (if `state == 'absent'`):
```suggestion
            gitlab_users_access.append({'name': gl_user, 'access_level': access_level_int[access_level] if access_level else None})
```